spec/authentication.yaml

```diff
@@ -132,70 +200,101 @@ components:
+            - [LDAP Authenticator](https://docs.conjur.org/Latest/en/Content/Integrations/ldap/ldap_authenticator.html) (`authn-ldap`)
+            - [Kubernetes Authenticator](https://docs.conjur.org/Latest/en/Content/Operations/Services/k8s_auth.htm) (`authn-k8s`)
+            - [Azure Authenticator](https://docs.conjur.org/Latest/en/Content/Operations/Services/azure_authn.htm) (`authn-azure`)
+
+          Note that the Kubernetes authenticator expects requests to this endpoint
+          to be sent with a trusted client certificate."
+        operationId: "getAccessTokenViaAuthenticator"
         parameters:
         - name: "authenticator"
           in: "path"
-          description: "The authenticator"
           required: true
+          description: "The authenticator"
           schema:
             type: string
             enum:
             - authn-iam
             - authn-ldap
             - authn-k8s
             - authn-azure
-          example: "authn-iam"
+          example: "authn-k8s"
         - name: "service_id"
           in: "path"
-          description: "The authenticator service"
           required: true
+          description: "URL-encoded authenticator service ID"
           schema:
             type: string
-          example: "okta"
+          example: "prod/okta"
         - name: "account"
           in: "path"
-          description: "Organization account name"
           required: true
+          description: "Organization account name"
           schema:
             $ref: 'openapi.yml#/components/schemas/AccountName'
         - name: "login"
           in: "path"
-          description: "Login name of the client. For users, it’s the user id. For hosts, the login name is `host/<host-id>`"
           required: true
+          description: "URL-encoded login name. For users, it’s the user ID. For hosts, the login name is `host/<host-id>`"
           schema:
             $ref: '#/components/schemas/LoginName'
         - name: "Accept-Encoding"
           in: "header"
           description: "Setting the Accept-Encoding header to base64 will return a pre-encoded access token"
           schema:
             type: "string"
+            enum:
+              - "application/json"
+              - "base64"
+            default: "application/json"
         requestBody:
-          description: "API Key"
-          required: true
+          description: "Authentication data as required by the specific authentication endpoint."
+          required: false
           content:
             text/plain:
               schema:
-                $ref: 'openapi.yml#/components/schemas/ApiKey'
+                oneOf:
+                  - $ref: 'openapi.yml#/components/schemas/ApiKey'
+                  - $ref: 'openapi.yml#/components/schemas/AWSSignatureHeader'
+            application/x-www-form-urlencoded:
+              schema:
+                $ref: 'openapi.yml#/components/schemas/AzureIdentityToken'
```

Comment:
@telday when the Python client is generated, it doesn't know what to do with this request body. it's only required for 3 of the authenticators it supports, so I had to move it to `required: false`. it shows like this in the method def:
```
:param UNKNOWN_BASE_TYPE unknown_base_type: Authentication data as required by the specific authentication endpoint.
```

do you know of a way to fix this? I can't tell if the problem is using `oneOf` for `text/plain` or if it's that `authn-k8s` doesn't require a body and I should split that authenticator out

Reply:
ahh I just tested without `oneOf` and it worked :/

Reply:
@izgeri Yeah currently although polymorphism is supported in OpenAPI in general, the API Generator does not [support](https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/python.md#schema-support-feature) it currently